src/main/frontend/src/components/AddReleaseYearForm.js

```diff
@@ -2,12 +2,146 @@
 // IMPORTANT:
 // You must update ResourceUrl.RESOURCES_VERSION each time whenever you're modified this file!
 //
+// @todo #1344 AddReleaseYearForm: edits by PR
 
 class AddReleaseYearForm extends React.Component {
-	
+	constructor(props) {
+		super(props);
+		this.state = {
+			year: null,
+			hasServerError: false,
+			validationErrors: [],
+			isDisabled: true,
+		};
+		this.handleSubmit = this.handleSubmit.bind(this);
+		this.handleChange = this.handleChange.bind(this);
+		this.calcRangYear = this.calcRangYear.bind(this);
+	}
+
+	calcRangYear(startYear, endYear) {
```

Comment:
Ok, I found that your solution is good enough :)

Only one moment here, let's rename it and remove the date-specific to make it more generic as it actually just generated a range of numbers:
```suggestion
	generateRange(start, end) {
```